crates/dprint/src/main.rs

```diff
@@ -40,6 +40,12 @@ fn main() {
 
 async fn run(runtime_handle: tokio::runtime::Handle) -> Result<(), AppError> {
   let args = arg_parser::parse_args(std::env::args().collect(), RealStdInReader)?;
+
+  #[cfg(feature = "completions_generator")]
+  if let arg_parser::SubCommand::GenerateCompletions = args.sub_command {
+    std::process::exit(0);
+  }
```

Comment:
Can we remove this custom section and then just have the subcommand handler below handle this? Doing the extra work below of setting things up is not a big deal.

Reply:
It will not print any thing to `stdout` while setting things up, right?

Reply:
It shouldn't. If so I can fix any issue later.